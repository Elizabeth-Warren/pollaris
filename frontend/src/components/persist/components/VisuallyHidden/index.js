import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Wrapper = styled.span`
  border: 0;
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;

  &:active,
  &:focus {
    ${({ focusable }) => focusable
      && css`
        clip: auto;
        clip-path: none;
        height: auto;
        margin: 0;
        overflow: visible;
        position: static;
        white-space: inherit;
        width: auto;
      `}
  }
`;

const VisuallyHidden = ({ children, focusable }) => (
  <Wrapper focusable={focusable}>{children}</Wrapper>
);

VisuallyHidden.propTypes = {
  children: PropTypes.node.isRequired,
  focusable: PropTypes.bool,
};

VisuallyHidden.defaultProps = {
  focusable: false,
};

export default VisuallyHidden;
