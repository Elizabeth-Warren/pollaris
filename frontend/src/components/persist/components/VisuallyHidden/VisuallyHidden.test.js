import React from 'react';
import renderer from 'react-test-renderer';
import VisuallyHidden from './index';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <VisuallyHidden>You can&apos;t see me.</VisuallyHidden>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
