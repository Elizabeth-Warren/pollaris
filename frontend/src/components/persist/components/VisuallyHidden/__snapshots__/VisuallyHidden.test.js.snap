// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<span
  className="c0"
  focusable={false}
>
  You can't see me.
</span>
`;
